packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -95,34 +95,53 @@ class Doctor {
     return buffer.toString();
   }
 
-  /// Print verbose information about the state of installed tooling.
-  Future<bool> diagnose({ bool androidLicenses: false }) async {
+  /// Print information about the state of installed tooling.
+  Future<bool> diagnose({ bool androidLicenses: false, bool verbose: true }) async {
     if (androidLicenses)
       return AndroidWorkflow.runLicenseManager();
 
     bool doctorResult = true;
+    int issues = 0;
 
     for (DoctorValidator validator in validators) {
       final ValidationResult result = await validator.validate();
 
-      if (result.type == ValidationType.missing)
+      if (result.type == ValidationType.missing ) {
         doctorResult = false;
+      }
+      if (result.type != ValidationType.installed) {
+        issues += 1;
+      }
 
       if (result.statusInfo != null)
         printStatus('${result.leadingBox} ${validator.title} (${result.statusInfo})');
       else
         printStatus('${result.leadingBox} ${validator.title}');
 
       for (ValidationMessage message in result.messages) {
-        final String text = message.message.replaceAll('\n', '\n      ');
-        if (message.isError) {
-          printStatus('    ✗ $text', emphasis: true);
-        } else {
-          printStatus('    • $text');
+        if (message.isError || message.isHint || verbose == true) {
+          final String text = message.message.replaceAll('\n', '\n      ');
+          if (message.isError) {
+            printStatus('    ✗ $text', emphasis: true);
+          } else if (message.isHint) {
+            printStatus('    ! $text');
+          } else {
+            printStatus('    • $text');
+          }
         }
       }
+      if (verbose)
+        printStatus('');
+    }
 
+    // Make sure there's always one line before the summary even when not verbose.
+    if (!verbose)
       printStatus('');
+    if (issues > 0) {
+      final String verboseSuggestion = verbose ? '.' : "; run 'flutter doctor -v' for details.";
```

Comment:
I guess what I was saying before was more that I don't know that it makes any more or less sense to suggest `-v` in the case where there _are_ errors than the case where there _aren't_, since the only thing `-v` will show you is things that aren't broken in either case.

Reply:
(Happy to defer to your better judgement though.)

Reply:
I agree. I originally suggested the current working, but that was when the categories were always "closed" when there were errors.

I think now it would make more sense to have the summary at the bottom just say "! Doctor found issues in 3 categories.", but then insert 'Doctor summary (to see all details, run flutter doctor -v)' at the top, i.e.:

```
$ bin/flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (on Linux, locale en_US.UTF-8, channel master)
[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[✓] Android Studio (version 3.0)
[✓] IntelliJ IDEA Community Edition (version 2017.2)
[!] IntelliJ IDEA Community Edition (version 2017.1)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
[!] IntelliJ IDEA Community Edition (version 2016.3)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ This install is older than the minimum recommended version of 2017.1.0.
[!] Connected devices
    ! No devices found

! Doctor found issues in 3 categories; 
```

Reply:
Done